modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java

```diff
@@ -606,32 +543,36 @@ public static void setHeaderValueStructType(BStruct struct) {
         headerValueStructType = struct.getType();
     }
 
-    public static void populateConnection(BStruct request, HTTPCarbonMessage cMsg) {
+    public static void populateConnection(BStruct request,
+            HTTPCarbonMessage cMsg) {
         request.addNativeData(Constants.TRANSPORT_MESSAGE, cMsg);
         request.setStringField(Constants.CONNECTION_HOST_INDEX,
                 ((InetSocketAddress) cMsg.getProperty(Constants.LOCAL_ADDRESS)).getHostName());
         request.setIntField(Constants.CONNECTION_PORT_INDEX, (Integer) cMsg.getProperty(Constants.LISTENER_PORT));
     }
 
-    public static void populateInboundRequest(BStruct request, HTTPCarbonMessage cMsg) {
+    public static void populateInboundRequest(BStruct request, BStruct entity, BStruct mediaType,
+            HTTPCarbonMessage cMsg) {
         request.addNativeData(Constants.TRANSPORT_MESSAGE, cMsg);
         request.addNativeData(Constants.INBOUND_REQUEST, true);
         request.setStringField(Constants.REQUEST_PATH_INDEX, (String) cMsg.getProperty(Constants.REQUEST_URL));
         request.setStringField(Constants.REQUEST_METHOD_INDEX, (String) cMsg.getProperty(Constants.HTTP_METHOD));
         request.setStringField(Constants.REQUEST_VERSION_INDEX, (String) cMsg.getProperty(Constants.HTTP_VERSION));
         Map<String, String> resourceArgValues = (Map<String, String>) cMsg.getProperty(Constants.RESOURCE_ARGS);
         request.setStringField(Constants.REQUEST_REST_URI_POSTFIX_INDEX,
-                resourceArgValues.get(Constants.REST_URI_POSTFIX));
+                                resourceArgValues.get(Constants.REST_URI_POSTFIX));
 
         if (cMsg.getHeader(Constants.USER_AGENT_HEADER) != null) {
             request.setStringField(Constants.REQUEST_USER_AGENT_INDEX, cMsg.getHeader(Constants.USER_AGENT_HEADER));
             cMsg.removeHeader(Constants.USER_AGENT_HEADER);
         }
-        request.setRefField(Constants.REQUEST_HEADERS_INDEX,
-                prepareHeaderMap(cMsg.getHeaders(), new BMap<>()));
+        populateEntity(entity, mediaType, cMsg);
+        request.addNativeData(MESSAGE_ENTITY, entity);
+        request.addNativeData(IS_ENTITY_BODY_PRESENT, false);
     }
 
-    public static void populateInboundResponse(BStruct response, HTTPCarbonMessage cMsg) {
+    public static void populateInboundResponse(BStruct response, BStruct entity, BStruct mediaType, HTTPCarbonMessage
```

Comment:
Shall we document these public methods for all possible cases?

Reply:
Fixed.